archaius2-api/src/main/java/com/netflix/archaius/api/PropertySource.java

```diff
@@ -0,0 +1,31 @@
+package com.netflix.archaius.api;
+
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+/**
+ * Contract for a raw source of properties
+ */
+public interface PropertySource {
+    /**
+     * Get the raw property value.  No interpolation or other modification is done to the property.
+     * @param key
+     */
+    default Optional<Object> getProperty(String key) { return Optional.empty(); }
+
+    /**
+     * Mechanism for consuming all properties of the PropertySource
+     * @param consumer
+     */
+    default void forEachProperty(BiConsumer<String, Object> consumer) {}
```

Comment:
Yes, we do.  This is used heavily in archaius to merge property sources and provides thread safety guarantees instead of having to externally lock getKeys() and call getProperty() when iterating a PropertySource.